compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -672,6 +690,62 @@ createProjectPackageDb opts fps = do
             , "--expand-pkgroot"
             ]
 
+    -- generate a package containing template instances and install it in the package database
+    generateAndInstallInstancesPkg ::
+           [(NormalizedFilePath, String)]
+        -> Options
+        -> FilePath
+        -> FilePath
+        -> String
+        -> String
+        -> String
+        -> Maybe String
+        -> [String]
+        -> IO ()
+    generateAndInstallInstancesPkg templInstSrc opts dbPathAbs projectPackageDatabaseAbs unitIdStr instancesUnitIdStr pkgName mbPkgVersion deps =
+        withTempDir $ \tempDir ->
+            withCurrentDirectory tempDir $ do
+                loggerH <- getLogger opts "generate instances package"
+                mapM_ writeSrc templInstSrc
+                sdkVersion <- getSdkVersion
+                let pkgConfig =
+                        PackageConfigFields
+                            { pName = "instances-" <> pkgName
+                            , pSrc = "."
+                            , pExposedModules = Nothing
+                            , pVersion = mbPkgVersion
+                            , pDependencies = (unitIdStr <.> "dalf") : deps
+                            , pSdkVersion = sdkVersion
+                            , cliOpts = Nothing
+                            }
+                opts' <-
+                    mkOptions $
+                    opts
+                        { optWriteInterface = True
+                        , optPackageDbs = projectPackageDatabaseAbs : optPackageDbs opts
+                        , optIfaceDir = Just "./"
+                        , optIsGenerated = True
+                        , optDflagCheck = False
+                        , optMbPackageName = Just instancesUnitIdStr
+                        , optHideAllPkgs = False
+                        , optPackageImports = [(unitIdStr, []) | pkgName /= "daml-stdlib"]
+                        }
+                mbDar <-
+                    withDamlIdeState opts' loggerH diagnosticsLogger $ \ide ->
+                        buildDar
+                            ide
+                            pkgConfig
+                            (toNormalizedFilePath $
+                             fromMaybe ifaceDir $ optIfaceDir opts')
+                            (FromDalf False)
+                dar <- mbErr "ERROR: Creation of instances DAR file failed." mbDar
+              -- TODO (drsk) switch to different zip library so we don't have to write
```

Comment:
I don’t understand this comment.

Reply:
we use two different zip libraries in Damlc. Here we write a zip archive generated with the first to disc and decode it with the second library. I want to just use one zip library in Damlc.hs

Reply:
Makes sense :+1: